datalad/support/gitrepo.py

```diff
@@ -1079,6 +1079,11 @@ def config(self):
     def is_with_annex(self):
         """Report if GitRepo (assumed) has (remotes with) a git-annex branch
         """
+        # OPT: To avoid "lengthy" invocation of git, first sample ref for
+        # `git-annex` branch
+        if op.exists(self.dot_git / "refs" / "heads" / "git-annex"):
```

Comment:
I expect that `dot_git` would be "correct" for bare repos ;)

Reply:
I don't know if the wink means you're being sarcastic, but just in case: it wouldn't work for bare repos (though it wouldn't do any harm either):

https://github.com/datalad/datalad/blob/1aebd29c563828e31121b2dc5d6b6da0110fb8b1/datalad/support/gitrepo.py#L628

It also wouldn't catch a packed git-annex ref (e.g., after a `git gc` run), though again that doesn't do any harm since we'd just fall back to the for-each.

Looks fine to me.

Reply:
> I don't know if the wink means you're being sarcastic, but just in case: it wouldn't work for bare repos (though it wouldn't do any harm either):
> 
> https://github.com/datalad/datalad/blob/1aebd29c563828e31121b2dc5d6b6da0110fb8b1/datalad/support/gitrepo.py#L628

I assumed that for the bare repos cases, the `.dot_git` value would be adjusted... to just point to the `.pathobj`. Was that too naive to assume?

packed refs: forgot about them.  Indeed an ideal solution here should also check `$GIT_DIR/packed-refs`.  Do you think it is worthwhile adding to this PR?

Reply:
> I assumed that for the bare repos cases, the `.dot_git` value would be adjusted... to just point to the `.pathobj`. Was that too naive to assume?

I don't spot the adjustment anywhere, but perhaps I'm missing it.

> packed refs: forgot about them. Indeed an ideal solution here should also check `$GIT_DIR/packed-refs`. Do you think it is worthwhile adding to this PR?

No, I don't think we should try to optimize this in this PR (or any other, really) by inspecting packed-refs.  I was just pointing out a case when this optimization wouldn't be in effect.